dracclient/resources/raid.py

```diff
@@ -98,15 +98,220 @@
+
+    def __init__(self, name, instance_id, current_value, pending_value,
+                 read_only, fqdd, min_length, max_length):
+        """Creates RAIDStringAttribute object
+
+        :param name: name of the RAID attribute
+        :param instance_id: InstanceID of the RAID attribute
+        :param current_value: current value of the RAID attribute
+        :param pending_value: pending value of the RAID attribute, reflecting
+                an unprocessed change (eg. config job not completed)
+        :param read_only: indicates whether this RAID attribute can be changed
+        :param fqdd: Fully Qualified Device Description of the RAID
+                Attribute
+        :param min_length: minimum length of the string
+        :param max_length: maximum length of the string
+        """
+        super(RAIDStringAttribute, self).__init__(name, instance_id,
+                                                  current_value,
+                                                  pending_value,
+                                                  read_only, fqdd)
+        self.min_length = min_length
+        self.max_length = max_length
+
+    @classmethod
+    def parse(cls, raid_attr_xml):
+        """Parses XML and creates RAIDStringAttribute object"""
+
+        raid_attr = RAIDAttribute.parse(cls.namespace, raid_attr_xml)
+        min_length = int(utils.get_wsman_resource_attr(
+            raid_attr_xml, cls.namespace, 'MinLength'))
+        max_length = int(utils.get_wsman_resource_attr(
+            raid_attr_xml, cls.namespace, 'MaxLength'))
+
+        return cls(raid_attr.name, raid_attr.instance_id,
+                   raid_attr.current_value, raid_attr.pending_value,
+                   raid_attr.read_only, raid_attr.fqdd, min_length, max_length)
+
+
 class RAIDManagement(object):
 
+    NAMESPACES = [(uris.DCIM_RAIDEnumeration, RAIDEnumerableAttribute),
+                  (uris.DCIM_RAIDString, RAIDStringAttribute)]
+
     def __init__(self, client):
         """Creates RAIDManagement object
 
         :param client: an instance of WSManClient
         """
         self.client = client
 
+    def list_raid_cntrl_settings(self, by_name=False):
+        """List the RAID configuration settings
+
+        :param by_name: Controls whether returned dictionary uses RAID
+                        attribute name or instance_id as key.
+        :returns: a dictionary with the RAID settings using InstanceID as the
+                  key. The attributes are either RAIDEnumerableAttribute,
+                  RAIDStringAttribute objects.
+        :raises: WSManRequestFailure on request failures
+        :raises: WSManInvalidResponse when receiving invalid response
+        :raises: DRACOperationFailed on error reported back by the DRAC
+                interface
+        """
+
+        return utils.list_settings(self.client, self.NAMESPACES, by_name)
+
+    def set_raid_cntrl_settings(self, new_settings, raid_fqdd):
```

Comment:
Can you change "cntrl" to "controller" in the method name above?

Reply:
Done